src/components/RangeSlider/components/DualThumb/DualThumb.tsx

```diff
@@ -106,6 +116,19 @@ export class DualThumb extends React.Component<DualThumbProps, State> {
     }
   }
 
+  componentDidUpdate() {
+    const {globalTheming} = this.state;
+    const unstableGlobalTheming = this.context
+      ? this.context.unstableGlobalTheming
+      : false;
+
+    if (unstableGlobalTheming !== globalTheming) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({globalTheming: unstableGlobalTheming});
+      this.setTrackPosition();
```

Comment:
Can you explain this change? I don't like seeing `setState` in `componentDidUpdate` and it also seems odd to have a state for `globalTheming`

Reply:
In most places where `unstableGlobalTheming` is being used, its within a functional component and we have access to the `useFeatures` hook. But `DualThumb` is a class based component, and even though you can access `this.context.unstableGlobalTheming` from the render section, this doesn't help change the size of the track because `setTrackPosition()` is what modifies the state values `trackWidth` and `trackLeft`.

So in order to call that function only whenever the theme changes, this is why I call it within a  `componentDidUpdate` only when this condition is true:

```
unstableGlobalTheming !== globalTheming
```

Reply:
I've pulled down this branch, and removed the componentDidUpdate life cycle method but can't seem to find any oddities while switching between current and new design language. Can you post a gif of the issue this is fixing?

Reply:
Not sure I follow either, if `this.context.newDesignLanguage` is true, it always returns true inside `setTrackPosition`. Not sure we need the state?

Reply:
hmm strange, I'm not having issues anymore when I get rid of `componentDidUpdate`. Thanks for challenging me on this!

Reply:
Awesome!